File: 0423/423.go
func reconstructOriginalDigitsFromE(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := reconstructOriginalDigitsFromE(root.Left)
	r := reconstructOriginalDigitsFromE(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0423/423.cs
public class Solution {
    public int ReconstructOriginalDigitsFromE(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(ReconstructOriginalDigitsFromE(root.left), ReconstructOriginalDigitsFromE(root.right));
    }
}